### ISSUE TITLE
How to update learning rate within the same epoch

### ISSUE THREAD

**Issue by bernardohenz** [junyanz/pytorch-CycleGAN-and-pix2pix#209]
Hi,

I am training use huge datasets, and my network starts giving acceptable results with just 4 epochs. I would like to apply learning rate decay, but the current implementation (using the ```python niter_decay```) does not feel appropriate for my case (as it only updates the lr once per epoch).

I would like to know if there is an easy way to use lr decay on this case, updating the lr more than once per epoch. Do I need to rewrite the ```python get_scheduler``` method?

Thanks for all.
Ps: this work is awesome :smiley:

**Comment by junyanz** [comment]
We haven't tried that. I guess you can move this [line](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/train.py#L57) into the inner [forloop](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/train.py#L22). You can implement your new learning rate policy with lambda function [here](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/models/networks.py#L88).

**Comment by junyanz** [comment]
It is not included in the official repo. But you can write your own.
